docs/blog/posts/2023-05-21-removing-pgp.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Removing PGP from PyPI.
+description: PyPI has removed support for PGP signatures.
+author: Donald Stufft
+publish_date: 2023-05-21
+date: "2023-05-21 00:00"
+tags:
+  - security
+---
+
+PyPI has removed support for uploading PGP signatures.
+
+If you are someone who is currently uploading signatures, your package uploads will
+continue to succeed, but any PGP signatures will be silently ignored. If you are
+someone who is currently downloading PGP signatures, existing signatures
+*SHOULD* continue to be available [^1], but no new signatures will be made available.
+The related API fields such as `has_sig` have all been hardcoded to always be
+`False`.
+
+Historically PyPI has supported uploading PGP signatures alongside the release
+artifacts in an attempt to provide some level of package signing. However, the
+approach used had long standing,
+[documented issues](https://caremad.io/posts/2013/07/packaging-signing-not-holy-grail/)
+which had lead us to demphasize the support
+for PGP signatures over time by removing them from the UI.
+
+PyPI continued to support uploading these signatures in the hope that there
+might be some systems out there that found them useful.
+
+However, a
+[recent look at the signatures on PyPI](https://blog.yossarian.net/2023/05/21/PGP-signatures-on-PyPI-worse-than-useless)
+shows us that the current support for PGP signatures is not proving useful.
+
+In the last 3 years, about 50k signatures had been uploaded to PyPI by 1069
+unique keys. Of those 1069 unique keys, about 30% of them were not discoverable
+on major public keyservers, making it difficult or impossible to meaningfully
+verify those signatures. Of the remaining 71%, nearly half of them were unable
+to be meaningfully verified at the time of the audit (2023-05-19) [^2].
+
+In other words, out of all of the unique keys that had uploaded signatures to
+PyPI, only 36% of them were capable of being meaningfully verified [^3] at the
+time of audit.
```

Comment:
Thirty-six percent sounds like a lot, I think we need an additional quantifier here, like "and those files that had verifiable signatures only represented 0.000X% of files on PyPI".

Reply:
To make it fair, I'll have to look up how many files have been released in the 3 year window that @woodruffw used in his blog post, but I can do that.

Reply:
There were 6556154 files without signatures and 53030 with them, so 0.3% if my math is right.

Reply:
I've added some wording to mention this.